packages/pigeon/run_tests.sh

```diff
@@ -86,195 +90,337 @@ test_pigeon_android() {
-# Make sure the artifacts are present.
-flutter precache
-# Make sure flutter dependencies are available.
-pushd $PWD
-cd e2e_tests/test_objc/
-flutter pub get
-popd
-test_pigeon_ios ./pigeons/async_handlers.dart
-test_null_safe_dart ./pigeons/message.dart
-test_pigeon_android ./pigeons/voidflutter.dart
-test_pigeon_android ./pigeons/voidhost.dart
-test_pigeon_android ./pigeons/host2flutter.dart
-test_pigeon_android ./pigeons/message.dart
-test_pigeon_android ./pigeons/void_arg_host.dart
-test_pigeon_android ./pigeons/void_arg_flutter.dart
-test_pigeon_android ./pigeons/list.dart
-test_pigeon_android ./pigeons/all_datatypes.dart
-test_pigeon_android ./pigeons/async_handlers.dart
-test_pigeon_ios ./pigeons/message.dart
-test_pigeon_ios ./pigeons/host2flutter.dart
-test_pigeon_ios ./pigeons/voidhost.dart
-test_pigeon_ios ./pigeons/voidflutter.dart
-test_pigeon_ios ./pigeons/void_arg_host.dart
-test_pigeon_ios ./pigeons/void_arg_flutter.dart
-test_pigeon_ios ./pigeons/list.dart
-test_pigeon_ios ./pigeons/all_datatypes.dart
+run_dart_compilation_tests() {
+  # Make sure the artifacts are present.
+  flutter precache
+  # Make sure flutter dependencies are available.
+  pushd $PWD
+  cd e2e_tests/test_objc/
+  flutter pub get
+  popd
+  test_pigeon_dart ./pigeons/all_datatypes.dart
+  test_pigeon_dart ./pigeons/async_handlers.dart
+  test_pigeon_dart ./pigeons/host2flutter.dart
+  test_pigeon_dart ./pigeons/list.dart
+  test_pigeon_dart ./pigeons/message.dart
+  test_pigeon_dart ./pigeons/void_arg_flutter.dart
+  test_pigeon_dart ./pigeons/void_arg_host.dart
+  test_pigeon_dart ./pigeons/voidflutter.dart
+  test_pigeon_dart ./pigeons/voidhost.dart
+}
 
-###############################################################################
-# iOS unit tests on generated code.
-###############################################################################
-pub run pigeon \
-  --no-dart_null_safety \
-  --input pigeons/message.dart \
-  --dart_out /dev/null \
-  --objc_header_out platform_tests/ios_unit_tests/ios/Runner/messages.h \
-  --objc_source_out platform_tests/ios_unit_tests/ios/Runner/messages.m
-pub run pigeon \
-  --no-dart_null_safety \
-  --input pigeons/async_handlers.dart \
-  --dart_out /dev/null \
-  --objc_header_out platform_tests/ios_unit_tests/ios/Runner/async_handlers.h \
-  --objc_source_out platform_tests/ios_unit_tests/ios/Runner/async_handlers.m
-clang-format -i platform_tests/ios_unit_tests/ios/Runner/messages.h
-clang-format -i platform_tests/ios_unit_tests/ios/Runner/messages.m
-clang-format -i platform_tests/ios_unit_tests/ios/Runner/async_handlers.h
-clang-format -i platform_tests/ios_unit_tests/ios/Runner/async_handlers.m
-pushd $PWD
-cd platform_tests/ios_unit_tests
-flutter build ios --simulator
-cd ios
-xcodebuild \
+run_java_compilation_tests() {
+  # Make sure the artifacts are present.
+  flutter precache
+  # Make sure flutter dependencies are available.
+  pushd $PWD
+  cd e2e_tests/test_objc/
```

Comment:
what's the significance of e2e_tests/test_objc/ in the java test? If there's a special pigeon schema file in that test, should it be moved somewhere more generic?

Reply:
It's for the aforementioned hack to get access to Flutter.   Added comment